File: app/src/main/java/com/alexkn/syntact/presentation/common/ListItemViewHolder.java
package com.alexkn.syntact.presentation.common;

import android.view.View;

import com.alexkn.syntact.data.common.Identifiable;

import androidx.recyclerview.widget.RecyclerView;

public abstract class ListItemViewHolder<T extends Identifiable> extends RecyclerView.ViewHolder {
    public ListItemViewHolder(View v) {
        super(v);
    }

    public abstract void bindTo(T entity);
}

File: app/src/main/java/com/alexkn/syntact/presentation/common/ListItemViewHolder.kt
package com.alexkn.syntact.presentation.common

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.alexkn.syntact.core.model.Identifiable

abstract class ListItemViewHolder<T : Identifiable<*>>(v: View) : RecyclerView.ViewHolder(v) {

    abstract fun bindTo(entity: T)
}
